Crash: concatenate different char-size strings with improper encoding

```c
/* Note: save with iso-8859-1 encoding */
int *p = L"" "ÿ";
```

```
panic: exact division produced remainder
/Users/ehaas/source/arocc/src/aro/Parser.zig:8083:67: 0x10e97dfe3 in stringLiteral (arocc)
    arr_ty.* = .{ .elem = string_kind.elementType(p.comp), .len = @divExact(slice.len, @intFromEnum(char_width)) };
                                                                  ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:7918:34: 0x10eb91600 in primaryExpr (arocc)
        => return p.stringLiteral(),
                                 ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:7290:40: 0x10eb85855 in unExpr (arocc)
                lhs = try p.primaryExpr();
                                       ^
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -7995,7 +7995,8 @@ fn stringLiteral(p: *Parser) Error!Result {
             return error.ParsingFailed;
         }
     }
-    assert(string_end > p.tok_i);
+    const count = string_end - p.tok_i;
+    assert(count > 0);
 
     const char_width = string_kind.charUnitSize(p.comp);
 
@@ -8041,7 +8042,13 @@ fn stringLiteral(p: *Parser) Error!Result {
                     },
                 }
             },
-            .improperly_encoded => |bytes| p.strings.appendSliceAssumeCapacity(bytes),
+            .improperly_encoded => |bytes| {
+                if (count > 1) {
+                    try p.errTok(.illegal_char_encoding_error, p.tok_i);
+                    return error.ParsingFailed;
+                }
+                p.strings.appendSliceAssumeCapacity(bytes);
+            },
             .utf8_text => |view| {
                 switch (char_width) {
                     .@"1" => p.strings.appendSliceAssumeCapacity(view.bytes),

--- a/test/cases/concat improperly encoded strings.c
+++ b/test/cases/concat improperly encoded strings.c
@@ -1,0 +1,5 @@
+/* Note: saved with iso-8859-1 encoding, do not change */
+int *p = L"" "ÿ";
+
+#define EXPECTED_ERRORS "concat improperly encoded strings.c:2:14: error: illegal character encoding in character literal" \
+

